common/connection.go

```diff
@@ -165,10 +182,13 @@ func NewConnection(ctx context.Context, wsURL string, logger *log.Logger) (*Conn
 		sessions:         make(map[target.SessionID]*Session),
 	}
 
+	return &c, nil
+}
+
+// start starts the connection's main control loop.
+func (c *Connection) start() {
```

Comment:
The `http` package is an entirely different case, though. It's a server and should run handlers in different goroutines. However, almost the entire standard library is synchronous. So, I'm not sure about the "plenty" part.

> What advantage does this bring in this scenario?

It doesn't bring much advantage in this case because our whole design needs to change. I wanted to make the first step in that direction. But I'll change this back until we agree on a better design.

Reply:
See https://github.com/grafana/xk6-browser/commit/e0d7db06098976fa646f39ef316fcbe081b4aeaf

Reply:
Thanks! Happy to discuss this, i'm interested in knowing more from your perspective.